src/Client.php

```diff
@@ -56,21 +56,53 @@ final class Client implements ClientInterface
      */
     private $integrations;
 
+    /**
+     * @var SerializerInterface|null The serializer of the client
+     */
+    private $serializer;
+
+    /**
+     * @var RepresentationSerializerInterface|null The representation serializer of the client
+     */
+    private $representationSerializer;
+
+    /**
+     * @var StacktraceBuilder
+     */
+    private $stacktraceBuilder;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    private $sdkIdentifier;
+
+    /**
+     * @var string the SDK version of the Client
+     */
+    private $sdkVersion;
+
     /**
      * Constructor.
      *
-     * @param Options               $options      The client configuration
-     * @param TransportInterface    $transport    The transport
-     * @param EventFactoryInterface $eventFactory The factory for events
-     * @param LoggerInterface|null  $logger       The PSR-3 logger
+     * @param Options                                $options                  The client configuration
+     * @param TransportInterface                     $transport                The transport
+     * @param string|null                            $sdkIdentifier            The Sentry SDK identifier
+     * @param string|null                            $sdkVersion               The Sentry SDK version
+     * @param SerializerInterface|null               $serializer               The serializer
+     * @param RepresentationSerializerInterface|null $representationSerializer The serializer for function arguments
+     * @param LoggerInterface|null                   $logger                   The PSR-3 logger
      */
-    public function __construct(Options $options, TransportInterface $transport, EventFactoryInterface $eventFactory, ?LoggerInterface $logger = null)
+    public function __construct(Options $options, TransportInterface $transport, ?string $sdkIdentifier, ?string $sdkVersion, ?SerializerInterface $serializer, ?RepresentationSerializerInterface $representationSerializer, ?LoggerInterface $logger = null)
```

Comment:
I would allow a `null` default for those additional dependencies
```suggestion
    public function __construct(
        Options $options,
        TransportInterface $transport,
        ?string $sdkIdentifier = null,
        ?string $sdkVersion = null,
        ?SerializerInterface $serializer = null,
        ?RepresentationSerializerInterface $representationSerializer = null,
        ?LoggerInterface $logger = null
    )
```